### PR TITLE
[nullserv] Add chart

### PR DESCRIPTION
**Pull request by brandon099**
Signed-off-by: Brandon Clifford <brandon@clifford.sh>

This adds a chart for `nullserv`, which can be used alongside DNS ad blocking software to speed up the responses since `nullserv` returns valid responses to the blocked/sinkholed requests.

[Upstream Project](https://github.com/bmrzycki/nullserv/)

**Benefits**

Supports installing `nullserv` to Kubernetes through a Helm chart.

**Additional information**

This PR currently depends on the `nullserv`  [Docker image](https://github.com/k8s-at-home/container-images/pull/121) to be made publicly available in the Github Container registry, which as of this writing is not yet available.

**Checklist** 
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Variables are documented in the README.md (this can be done with using our helm-docs wrapper `./hack/gen-helm-docs.sh stable <chart>`)

### CODE REVIEW

**Review by nicholaswilde — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! 🙏🏽

**Comment by bjw-s** [comment]
I've just made the underlying image available publicly, will do the PR review in a bit

File: charts/incubator/nullserv/values.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+image:
+  # -- image repository
+  repository: ghcr.io/k8s-at-home/nullserv
+  # -- image tag
+  tag: 1.3.0
+  # -- image pull policy
+  pullPolicy: IfNotPresent
+
+    # -- environment variables. See more environment variables in the [nullserv documentation](https://github.com/bmrzycki/nullserv/blob/master/README.md).
```
Comment (bjw-s) [inline charts/incubator/nullserv/values.yaml:20]:
```suggestion
# -- environment variables. See more environment variables in the [nullserv documentation](https://github.com/bmrzycki/nullserv/blob/master/README.md).
```

File: charts/incubator/nullserv/values.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ports:
+      http:
+        port: 80
+  https:
+    enabled: true
+    type: NodePort
+    ports:
+      https:
+        enabled: true
+        port: 443
```
Comment (bjw-s) [inline charts/incubator/nullserv/values.yaml:41]: Does this have to be a separate service, or can it also be a second port on the main service?

File: charts/incubator/nullserv/values.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+ingress:
+  # -- Enable and configure ingress settings for the chart under this key.
+  # @default -- See values.yaml
+  main:
+    enabled: false
+
+probes:
```
Comment (bjw-s) [inline charts/incubator/nullserv/values.yaml:49]:
```suggestion
# -- Configures the probes for the main Pod.
# @default -- See values.yaml
probes:
```

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: charts/incubator/nullserv/values.yaml (hunk 2 above)
Comment (brandon099) [reply charts/incubator/nullserv/values.yaml:41]: It does not have to be a separate service, and I would prefer it be one service with on port, but I couldn't tell how to add another port to the main service in the common v3 library chart. I'll look more into that.

Comment (brandon099) [reply charts/incubator/nullserv/values.yaml]: Well, that was easy. Now it is one service with multiple ports.

**Comment by nicholaswilde** [comment]
> I've just made the underlying image available publicly, will do the PR review in a bit

I'm not able to pull the image and it looks like the ci is not able to pull it as well.

**Comment by bjw-s** [comment]
> > I've just made the underlying image available publicly, will do the PR review in a bit
> 
> 
> 
> I'm not able to pull the image and it looks like the ci is not able to pull it as well.

We've had some reports of GitHub Container Registry having some issues today, this might be related. I'll kick the CI and see what happens.

File: charts/stable/nullserv/values.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+image:
+  # -- image repository
+  repository: ghcr.io/k8s-at-home/nullserv
+  # -- image tag
+  tag: 1.3.0
+  # -- image pull policy
+  pullPolicy: IfNotPresent
+
+    # -- environment variables. See more environment variables in the [nullserv documentation](https://github.com/bmrzycki/nullserv/blob/master/README.md).
```
Comment (bjw-s) [inline charts/stable/nullserv/values.yaml:20]:
```suggestion
# -- environment variables. See more environment variables in the [nullserv documentation](https://github.com/bmrzycki/nullserv/blob/master/README.md).
```

**Review by bjw-s — DISMISSED** [review]

**Review by bjw-s — APPROVED** [review]
After prefixing the tag with a `v` everything looks good now. Thanks a lot for submitting the PR!

**Comment by bjw-s** [comment]
@all-contributors please add @brandon099  for code

**Comment by brandon099** [comment]
Looks like the GitHub action to publish the chart failed and was not re-triggered for some reason. So even though it's merged, it's not available to install via the Helm repo.

**Comment by bjw-s** [comment]
Sorry about that! Looks like it got fixed automagically because of the merge of another chart just now.

**Pull request merged**